From da0c66bcddbe4e6ebc72357c9f6c5de75e176744 Mon Sep 17 00:00:00 2001
From: dundargoc <gocdundar@gmail.com>
Date: Sun, 5 Mar 2023 17:06:13 +0100
Subject: [PATCH] build: remove workaround for incorrectly packaged libluv

This removes a workaround for incorrectly packaged libluv in
90e44ecf1144cb32195da00e24d23afb111ea680 as it should not be needed
anymore.
---
 cmake/FindLibluv.cmake  | 11 +----------
 src/nvim/CMakeLists.txt |  8 ++------
 2 files changed, 3 insertions(+), 16 deletions(-)

diff --git a/src/nvim/CMakeLists.txt b/src/nvim/CMakeLists.txt
index 61530f5a7bf95..51e0727cd51c4 100755
--- a/src/nvim/CMakeLists.txt
+++ b/src/nvim/CMakeLists.txt
@@ -14,12 +14,8 @@ else()
 # TODO(dundargoc): unittest stops working if I create an pseudo-imported
 # library "luv" as with the other dependencies. Figure out why and fix.
 find_package(Luv 1.43.0 REQUIRED)
-target_include_directories(main_lib SYSTEM BEFORE INTERFACE ${LUV_INCLUDE_DIRS})
-# Use "luv" as imported library, to work around CMake using "-lluv" for
-# "luv.so".  #10407
-add_library(luv UNKNOWN IMPORTED)
-set_target_properties(luv PROPERTIES IMPORTED_LOCATION ${LUV_LIBRARY})
-target_link_libraries(main_lib INTERFACE luv)
+target_include_directories(main_lib SYSTEM BEFORE INTERFACE ${LUV_INCLUDE_DIR})
+target_link_libraries(main_lib INTERFACE ${LUV_LIBRARY})
 
 find_package(Iconv REQUIRED)
 find_package(Libuv 1.28.0 REQUIRED)
